vault/logical_system.go

```diff
@@ -4273,21 +4275,30 @@ func (b *SystemBackend) pathInternalUIMountsRead(ctx context.Context, req *logic
 			continue
 		}
 
-		if ns.ID == entry.NamespaceID && hasAccess(ctx, entry) {
-			if isAuthed {
-				// If this is an authed request return all the mount info
-				authMounts[entry.Path] = b.mountInfo(ctx, entry)
-			} else {
-				authMounts[entry.Path] = map[string]interface{}{
-					"type":        entry.Type,
-					"description": entry.Description,
-					"options":     entry.Options,
+		if ns.ID == entry.NamespaceID {
+
+			// filters UI login method type dropdown when a user is unauthenticated
+			if entry.Config.ListingVisibility == ListingVisibilityDropdown && !strutil.StrListContains(authTypes, entry.Type) {
```

Comment:
Thank you for calling this out. This is my first attempt at writing go and didn't feel particularly efficient to loop each time - but admittedly this was I had no idea how to otherwise approach! 

Will amend